packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx

```diff
@@ -13,34 +15,149 @@ import Text from '@popup/popupX/shared/Text';
+import { useAtomValue } from 'jotai';
+import { credentialsAtom } from '@popup/store/account';
+import { WalletCredential } from '@shared/storage/types';
+import { useIdentityName, useWritableSelectedAccount } from '@popup/shared/utils/account-helpers';
+import { useAccountInfo } from '@popup/shared/AccountInfoListenerContext';
+import { displayAsCcd } from 'wallet-common-helpers';
 
-const ACCOUNT_LIST = [
-    {
-        account: 'Account 1',
-        address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh',
-        balance: '4,227.38',
-        attached: 'Identity 1',
-    },
-    {
-        account: 'Account 2',
-        address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh',
-        balance: '4,227.38',
-        attached: 'Identity 1',
-    },
-    {
-        account: 'Account 3',
-        address: 'tt2kgdygjrsqtzq2n0yrf2493p83kkfjh50eo',
-        balance: '1,195.41',
-        attached: 'Identity 2',
-    },
-];
+function fallbackAccountName(credentialNumber: number): string {
+    return `Account ${1 + credentialNumber}`;
+}
 
-export default function Accounts() {
+type EditableAccountNameProps = {
+    currentName: string;
+    fallbackName: string;
+    onNewName: (newName: string) => void;
+};
+
+function EditableAccountName({ currentName, fallbackName, onNewName }: EditableAccountNameProps) {
+    const [isEditingName, setIsEditingName] = useState(false);
+    const [editedName, setEditedName] = useState(currentName);
+    // Using editedName instead of currentName to avoid flickering after completing.
+    const displayName = editedName === '' ? fallbackName : editedName;
+    const onAbort = () => {
+        setIsEditingName(false);
+        setEditedName(currentName);
+    };
+    const onComplete = () => {
+        onNewName(editedName.trim());
+        setIsEditingName(false);
+    };
+    const onEdit = () => {
+        setEditedName(currentName);
+        setIsEditingName(true);
+    };
+    const onInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setEditedName(event.target.value);
+    };
+    const onKeyUp = (event: KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === 'Enter') {
+            event.preventDefault();
+            onComplete();
+        }
+    };
+    if (isEditingName) {
+        return (
+            <>
+                <Text.Main>
+                    <input
+                        autoFocus
+                        className="editable"
+                        value={editedName}
+                        placeholder={fallbackName}
+                        onChange={onInputChange}
+                        onKeyUp={onKeyUp}
+                        maxLength={25}
+                    />
+                </Text.Main>
+                <Button.Icon className="transparent" icon={<Checkmark />} onClick={onComplete} />
+                <Button.Icon className="transparent" icon={<Close />} onClick={onAbort} />
+            </>
```

Comment:
I would suggest something like
```suggestion
            <>
                <Text.Main>
                    <input
                        autoFocus
                        className="editable"
                        value={editedName}
                        placeholder={fallbackName}
                        onChange={onInputChange}
                        onKeyUp={onKeyUp}
                        maxLength={25}
                    />
                </Text.Main>
                <div className="flex-row">
                    <Button.Icon className="transparent m-r-20" icon={<Checkmark className="checkmark-icon" />} onClick={onComplete} />
                    <Button.Icon className="transparent" icon={<Close />} onClick={onAbort} />
                </div
            </>
```

Reply:
I have done something like this now